Configure polling and marathon event callback

Grant1219 opened the pull request:
These two changes are meant to make it easier to use Bamboo along with Envoy, and to prevent Bamboo from rapid reloading Envoy which causes it to crash.

The build works:
```
> docker build -t bamboo-build -f Dockerfile-deb .
.....
Successfully built e689988894b2
Successfully tagged bamboo-build:latest
```

```
> docker run --rm -v $(pwd)/output:/output -e "_BAMBOO_VERSION=1.3.4" bamboo-build
/opt/go/src/github.com/seomoz/roger-bamboo/builder/build /opt/go/src/github.com/seomoz/roger-bamboo/builder
/opt/go/src/github.com/seomoz/roger-bamboo/builder/build/bamboo /opt/go/src/github.com/seomoz/roger-bamboo/builder/build /opt/go/src/github.com/seomoz/roger-bamboo/builder
{:timestamp=>"2019-03-19T18:21:10.151603+0000", :message=>"Debian packaging tools generally labels all files in /etc as config files, as mandated by policy, so fpm defaults to this behavior for deb packages. You can disable this default behavior with --deb-no-default-config-files flag", :level=>:warn}
{:timestamp=>"2019-03-19T18:21:10.623400+0000", :message=>"Created package", :path=>"bamboo_1.3.4-1_all.deb"}
/opt/go/src/github.com/seomoz/roger-bamboo/builder/build /opt/go/src/github.com/seomoz/roger-bamboo/builder
> ls output/
bamboo_1.3.4-1_all.deb
```

Grant1219 committed:
Add PollingInterval to Bamboo configuration

Being able to control the polling interval will allow Bamboo to fetch
updates more quickly than once every 30 seconds, but also keep updates
on a consistent cadance instead of triggering a reload on every Marathon
change event.

Reducing the frequency of Bamboo reloads will keep Envoy from reloading
too quickly and potentially crashing.

Grant1219 committed:
Make the Marathon event callback optional

By turning off the registration of the Marathon event callback, Bamboo
can be put into polling-only mode where it only fetches changes from
Marathon on a set interval, instead of possibly triggering several
reloads within a few seconds when events are received from Marathon.

This option will make it easier to control when reloads happen when
using Bamboo along with Envoy proxy.

Hosweay reviewed (approved):
Looks good. My only nit would be to make the version 1.4.0 and not 1.3.4

Grant1219 committed:
Update example configs

Since I added the PollingInterval and RegisterCallback options, these
need to be added to the example configs so they stay valid.

Grant1219 commented:
Closing this since the Envoy changes are being rolled back for now. If we can configure Envoy via the xDS APIs then Bamboo won't even be needed anymore.
